modin/backends/pandas/query_compiler.py

```diff
@@ -1154,12 +1154,17 @@ def unique(self):
     dt_days = MapFunction.register(_dt_prop_map("days"))
     dt_microseconds = MapFunction.register(_dt_prop_map("microseconds"))
     dt_nanoseconds = MapFunction.register(_dt_prop_map("nanoseconds"))
-    dt_components = MapFunction.register(_dt_prop_map("components"))
     dt_qyear = MapFunction.register(_dt_prop_map("qyear"))
     dt_start_time = MapFunction.register(_dt_prop_map("start_time"))
     dt_end_time = MapFunction.register(_dt_prop_map("end_time"))
     dt_to_timestamp = MapFunction.register(_dt_func_map("to_timestamp"))
 
+    def dt_components(self):
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            1, lambda df: df.squeeze(axis=1).dt.components, new_index=self.index,
+        )
+        return self.__constructor__(new_modin_frame)
+
```

Comment:
Could it also be implemented with `MapFunction` since `map` now has validation for changing internal indices and columns?

Reply:
Result of `dt.components` is  new `DataFrame`. Number of columns is changing. In this case we cannn't apply `MapFunction`.

Reply:
I think approach with `MapFunction` would work well if we'll add columns updating in a `_map` function at `modin_frame`

Reply:
@dchigarev Yes, this was added. Review, please.

Reply:
thanks